scene/2d/physics_body_2d.cpp

```diff
@@ -1389,6 +1398,18 @@ void CharacterBody2D::_move_and_slide_floating(double p_delta) {
 
 		first_slide = false;
 	}
+
+	// Scales the velocity according to the wall slope.
+	if (is_on_wall() && velocity.dot(motion_results.get(0).collision_normal) < 0) {
+		// Slide the velocity against the first collision.
+		Vector2 slide_motion = velocity.slide(motion_results.get(0).collision_normal);
+		if (velocity.dot(slide_motion) < 0) {
```

Comment:
I think this is probably mathematically impossible, but I put the check in anyway. It could be removed if deemed pointless.